icu4c/source/i18n/messageformat2_macros.h

```diff
@@ -46,6 +46,17 @@ using namespace pluralimpl;
 #define LOWERCASE_E ((UChar32)0x0065)
 #define UPPERCASE_E ((UChar32)0x0045)
 
+#define DIGIT_ZERO ((UChar32)0x0030)
+#define DIGIT_ONE ((UChar32)0x0031)
+#define DIGIT_TWO ((UChar32)0x0032)
```

Comment:
why do these have 'DIGIT'?

Reply:
The `DIGIT_` was to work around an error where an `enum` with `ZERO`, `ONE`, and `TWO` as variants is defined in `standardplural.h`, which was included indirectly in `messageformat2_function_registry.cpp` through `number_types.h`, and the messageformat2 macros were getting applied to the enum definition and confusing the compiler. However, it turns out the include of `number_types.h` wasn't needed anymore, so I'm removing it.

Reply:
Never mind -- it's still necessary because `itformat.cpp` in the test suite indirectly includes both `standardplural.h` and `messageformat2_macros.h`.

Reply:
Why do we need macros like this? Why not use `u'0'`, `u'1'`, ...?

Reply:
More generally, preprocessor macros with generic names in a public header file are a bad idea. We should use regular C++ character and string literals, or C++ constants in an appropriate namespace, or, if we really need macros, use a UXYZ_ prefix, maybe UPRV_ for internal stuff.

And anything that's in a public header file and not in a private class section needs an API doc comment, maybe `/** @internal */`.

Reply:
that's a better solution. But perhaps they aren't needed at all.

Reply:
Eliminated these in [1bdaa31](https://github.com/unicode-org/icu/pull/3063/commits/1bdaa3191bbc894d9ed3799c7f944c70216fb53f)